### ISSUE TITLE
JDL generation Bug 

### ISSUE THREAD

**Issue by MHTaleb** [jhipster/generator-jhipster#9212]
##### **Overview of the issue**

Hi
The issue is that the generated Seance.java class have scaffolding errors where it adds characters to the class as in the picture below
[image](https://i.stack.imgur.com/LWuzN.png)
##### **Motivation for or Use Case**

Really I dont know why this is happening, but if I believe that this is a bug because the S() symbol have nothing to do with my jdl, If it was only titre I would say I m at fault because I m using this keyword in my jdl. but since there is the S() I m sure that this is a bug somewhere and beond my level because I m learning about jhipster and still learning about spring 
##### **Reproduce the error**

easy:

1. generate a jhipster project using the same cofig as mine
2. use the : jhipster import-jdl yourjdl.jh
3. go check in src/domain the Seance.Java in line 147 and 153

##### **Related issues**

no, I searched but didn't found it , is not a duplicate since I first asked on 
[stackoverflow](https://stackoverflow.com/questions/54564283/why-my-jdl-file-is-generating-domains-with-error-in-code?noredirect=1#comment95937518_54564283)
##### **Suggest a Fix**

If I could I would but sorry this is beond my current level

##### **JHipster Version(s)**

```
boxing@0.0.0 C:\Users\win 10\Documents\salle de sport
`-- generator-jhipster@5.8.1

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
    "generator-jhipster": {
        "promptValues": {
            "packageName": "com.talcorp.boxing",
            "nativeLanguage": "fr"
        },
        "jhipsterVersion": "5.8.1",
        "applicationType": "monolith",
        "baseName": "boxing",
        "packageName": "com.talcorp.boxing",
        "packageFolder": "com/talcorp/boxing",
        "serverPort": "8080",
        "authenticationType": "jwt",
        "cacheProvider": "ehcache",
        "enableHibernateCache": true,
        "websocket": false,
        "databaseType": "sql",
        "devDatabaseType": "h2Disk",
        "prodDatabaseType": "mariadb",
        "searchEngine": false,
        "messageBroker": false,
        "serviceDiscoveryType": false,
        "buildTool": "maven",
        "enableSwaggerCodegen": false,
        "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
        "clientFramework": "angularX",
        "useSass": true,
        "clientPackageManager": "npm",
        "testFrameworks": ["gatling", "cucumber", "protractor"],
        "jhiPrefix": "jhi",
        "entitySuffix": "",
        "dtoSuffix": "DTO",
        "otherModules": [],
        "enableTranslation": true,
        "nativeLanguage": "fr",
        "languages": ["fr", "ar-ly"]
    }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Club {
  federation String,
  nomClub String,
  dateFondation LocalDate
}
entity Boxeur {
  nom String,
  prenom String,
  dateNaissance LocalDate,
  lieuNaissance String,
  dateInscription LocalDate
}
entity Photo {
  photo ImageBlob
}
entity Entraineur {
  nom String,
  prenom String,
  dateNaissance LocalDate,
  lieuNaissance String,
  president Boolean
}
entity Seance {
  titre String,
  detail String,
  dateSeance LocalDate,
  lieu LieuSeance,
  typeSeance TypeSeance
}
entity Payement {
  montant Long,
  datePayement LocalDate
}

enum LieuSeance {
  SALLE,
  STADE,
  MONTAGNE,
  AUTRE
}

enum TypeSeance {
  COMPETITION,
  SPARRING,
  ENDURANCE,
  STREATCHING,
  TECHNIQUE,
  PHYSIQUE,
  ENTRETIEN,
  AUTRE
}

relationship OneToOne {
  Club{president(nom)} to Entraineur,
  Photo{entraineur} to Entraineur{photo},
  Photo{boxeur} to Boxeur{photo}
}
relationship OneToMany {
  Entraineur{seances} to Seance{entreneur(nom)},
  Boxeur{versement} to Payement{adherant(nom)}
}
relationship ManyToMany {
  Seance{participants(nom)} to Boxeur{assistes}
}

dto Club, Boxeur, Photo, Entraineur, Seance, Payement with mapstruct
paginate Club, Boxeur, Photo, Entraineur, Seance, Payement with pagination
service Club, Boxeur, Photo, Entraineur, Seance, Payement with serviceImpl

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)

git version 2.19.1.windows.1

node: v10.14.1

npm: 6.7.0

yeoman: 2.0.5

Docker version 18.09.1, build 4c52b90

docker-compose version 1.23.2, build 1110ad01

INFO! Congratulations, JHipster execution is complete!

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

since the error is  in Seance and there is some code where Boxeur is there I will add the two entities
<details>
<summary>Seance.json file</summary>
<pre>
{
    "name": "Seance",
    "fields": [
        {
            "fieldName": "titre",
            "fieldType": "String"
        },
        {
            "fieldName": "detail",
            "fieldType": "String"
        },
        {
            "fieldName": "dateSeance",
            "fieldType": "LocalDate"
        },
        {
            "fieldName": "lieu",
            "fieldType": "LieuSeance",
            "fieldValues": "SALLE,STADE,MONTAGNE,AUTRE"
        },
        {
            "fieldName": "typeSeance",
            "fieldType": "TypeSeance",
            "fieldValues": "COMPETITION,SPARRING,ENDURANCE,STREATCHING,TECHNIQUE,PHYSIQUE,ENTRETIEN,AUTRE"
        }
    ],
    "relationships": [
        {
            "relationshipType": "many-to-many",
            "otherEntityRelationshipName": "assistes(titre)",
            "relationshipName": "participants",
            "otherEntityName": "boxeur",
            "otherEntityField": "nom",
            "ownerSide": true
        },
        {
            "relationshipType": "many-to-one",
            "relationshipName": "entreneur",
            "otherEntityName": "entraineur",
            "otherEntityField": "nom",
            "otherEntityRelationshipName": "seances"
        }
    ],
    "changelogDate": "20190207164554",
    "entityTableName": "seance",
    "dto": "mapstruct",
    "pagination": "pagination",
    "service": "serviceImpl",
    "jpaMetamodelFiltering": false,
    "fluentMethods": true,
    "clientRootFolder": "",
    "applications": "*"
}
</pre>
</details>

<details>
<summary>Boxeur.json file</summary>
<pre>
{
    "name": "Boxeur",
    "fields": [
        {
            "fieldName": "nom",
            "fieldType": "String"
        },
        {
            "fieldName": "prenom",
            "fieldType": "String"
        },
        {
            "fieldName": "dateNaissance",
            "fieldType": "LocalDate"
        },
        {
            "fieldName": "lieuNaissance",
            "fieldType": "String"
        },
        {
            "fieldName": "dateInscription",
            "fieldType": "LocalDate"
        }
    ],
    "relationships": [
        {
            "relationshipType": "one-to-many",
            "relationshipName": "versement",
            "otherEntityName": "payement",
            "otherEntityRelationshipName": "adherant"
        },
        {
            "relationshipType": "one-to-one",
            "relationshipName": "photo",
            "otherEntityName": "photo",
            "ownerSide": false,
            "otherEntityRelationshipName": "boxeur"
        },
        {
            "relationshipType": "many-to-many",
            "relationshipName": "assistes",
            "otherEntityName": "seance",
            "ownerSide": false,
            "otherEntityRelationshipName": "participants"
        }
    ],
    "changelogDate": "20190207164551",
    "entityTableName": "boxeur",
    "dto": "mapstruct",
    "pagination": "pagination",
    "service": "serviceImpl",
    "jpaMetamodelFiltering": false,
    "fluentMethods": true,
    "clientRootFolder": "",
    "applications": "*"
}
</pre>
</details>
##### **Browsers and Operating System**

I m on Windows 10, this bug is no browser related
-   [+] Checking this box is mandatory (this is just to show you read everything)

**Comment by ruddell** [comment]
Duplicate of https://github.com/jhipster/generator-jhipster/issues/9202

**Comment by pascalgrimaud** [comment]
Thanks @ruddell
Because of this issue, maybe we need to do a patch release ?
But I don't know if we revert or not the pull request about yeoman API

**Comment by ruddell** [comment]
We will need both a release of JHipster Core and the generator on the v5 branch.  I'm not sure which needle pull request is the one for fixing v5 method names, @avdev4j would know for sure

I'm curious how this wasn't picked up by the CI testing since it should cause the Java to fail compilation, haven't had time to look into it though.

**Comment by pvliss** [comment]
> I'm curious how this wasn't picked up by the CI testing since it should cause the Java to fail compilation

@ruddell I believe the current test suite misses that edge case. I can submit a PR that only changes the `test-integration/samples/jdl-default/app.jdl` file and will cause the CI to fail or even better use `https://github.com/pvliss/jhipster-core/blob/64786e8429b68e3ff0fbfd6384723340ca205456/test/test_files/big_sample.jdl` instead to cover even more test cases.

Actually, there seem to be some more edge cases missed by the jhipster-core tests as well and as I have mentioned in this [comment](https://github.com/jhipster/jhipster-core/pull/304#issuecomment-461309089) I would like to make a PR to add missing edge cases and then to slightly refactor the code as it contains a lot of duplication and is a bit hard to read, test and maintain.

**Comment by pascalgrimaud** [comment]
@pvliss : yes plz, go ahead. I coded the CI with JDL part, but didn't take the time to test a real JDL. So your suggestion to replace this app.jdl is good.

I plan to do a specific builds with only JDL at hipster-labs

**Comment by avdev4j** [comment]
@ruddell 
For needle API we have two open PR, one for the master branch (#9151) and one for the v5 one (#9214).
Same commits and same modifications. (other related PR have been closed because they now are redundant).

@pvliss 
My work on needle API was made to have a more readable and maintainble code. This bug was finally useful to highlight a lack of test case (It was not really intended :D).
I plan to add more tests but I lack of time :(. If you want to do a PR it will be really appreciate and maybe I can help you if you need.

**Comment by pvliss** [comment]
@pascalgrimaud : I have added the PR https://github.com/jhipster/generator-jhipster/pull/9217. 

@avdev4j: Thanks, but this is not related to #9151. This is a regression in jhipster-core from one of my changes https://github.com/jhipster/jhipster-core/pull/302 that has already been fixed in https://github.com/jhipster/jhipster-core/pull/304. More tests need to be written at jhipster-core as well. The problem though is that it could have also been caught in the CI of the main generator but didn't as test case here is simple as well and we are looking into rectifying that as well
